tests/XRootD/scitokens.sh

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env bash
+
+function setup_scitokens() {
+	require_commands curl
+	if [ -z "$HOSTNAME" ]; then
+		HOSTNAME=`hostname`
+	fi
+
+	cat > "$PWD/scitokens/authdb" << EOF
+u * /issuer lr
+EOF
+	cat > "$PWD/scitokens/scitokens.cfg" <<EOF
+[Global]
+audience = https://$HOSTNAME:8095
+
+[Issuer test]
+issuer = https://$HOSTNAME:8095/issuer/one
+base_path = /protected
+EOF
+
+	# Setup the issuer and protected directory
+	ln -sf "$PWD/../issuer/export" "$PWD/scitokens/xrootd/issuer"
+	mkdir -p "$PWD/scitokens/xrootd/protected"
+	echo 'Hello, World' > "$PWD/scitokens/xrootd/protected/hello_world.txt"
+
+	# Override the scitoken cache location; otherwise, contents of prior test runs may be cached
+	export XDG_CACHE_HOME=$(mktemp -d "${PWD}/${NAME}/cache-XXXXXX")
+	mkdir -p "$XDG_CACHE_HOME/scitokens"
+
+	# Create a read-only token
+	$PWD/../scitokens/xrdscitokens-create-token ../issuer/issuer_pub_1.pem ../issuer/issuer_key_1.pem test_1 https://$HOSTNAME:8095/issuer/one storage.read:/ > "$PWD/scitokens/token"
+	if [ "$?" -ne 0 ]; then
+		echo "Failed to create token"
+		exit 1
+	fi
+
+	# Create a create-only token
+	$PWD/../scitokens/xrdscitokens-create-token ../issuer/issuer_pub_1.pem ../issuer/issuer_key_1.pem test_1 https://$HOSTNAME:8095/issuer/one storage.create:/subdir > "$PWD/scitokens/token_create"
+	if [ "$?" -ne 0 ]; then
+		echo "Failed to create 'storage.create' token"
+		exit 1
+	fi
+
+	# Create a modify token
+	$PWD/../scitokens/xrdscitokens-create-token ../issuer/issuer_pub_1.pem ../issuer/issuer_key_1.pem test_1 https://$HOSTNAME:8095/issuer/one storage.modify:/subdir_modify > "$PWD/scitokens/token_modify"
+	if [ "$?" -ne 0 ]; then
+		echo "Failed to create 'storage.modify' token"
+		exit 1
+	fi
+}
+
+function teardown_scitokens() {
+	echo
```

Comment:
You don't have to define a function if it's not doing anything. It's run [conditionally](https://github.com/xrootd/xrootd/blob/8a4ca755c57bd01b807a74a4aa3fc89b3f42116b/tests/XRootD/test.sh#L157-L159), if it exists.

Reply:
Removed in latest rebase.